File: Java/003.无重复字符的最长子串.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> map = new HashMap<>();
        int res = 0,left = -1;
        for(int i = 0; i < s.length(); i++){
            if(map.containsKey(s.charAt(i)) && map.get(s.charAt(i)) > left){
                left = map.get(s.charAt(i));
            }
            res = Math.max(res, i - left);
            map.put(s.charAt(i),i);
        }
        return res;
    }
}

File: CPP/003.无重复字符的最长子串.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int,int> mp;
        int n = s.size();
        int i = 0, left = -1, res = 0;
        for(int i =0; i < n; i++){
            if(mp.count(s[i]) && mp[s[i]]>left){
                left = mp[s[i]];
            }
            res = max(res, i - left);
            mp[s[i]] = i;
        }
        return res;
    }
};